extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+
+
+def licenses(id):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if id is None:
+        return []
+    else:
+        return [{"license": {
+            "id": id
+        }}]
+
+
+def package_url(node: dict) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    qualifiers = {
+        "prev": node["prev"]
+    }
+    if node["user"]:
+        qualifiers["user"] = node["user"]
+    if node["channel"]:
+        qualifiers["channel"] = node["channel"]
+    if "#" in node["ref"]:
+        qualifiers["rref"] = node["ref"].split("#")[1]
+    qualifiers["repository_url"] = node["remote"] if node["remote"] else "https://center.conan.io"
+    return PackageURL(
+        type="conan",
+        name=node["name"],
+        version=node["version"],
+        qualifiers=qualifiers)
+
+
+def component(n: dict) -> dict:
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components
+    """
+    result = {
+        "type": "library",
+        "bom-ref": n["id"],
```

Comment:
While the only formal requirement on the bom-ref is that it is unique within an SBOM, I'd personally prefer a more expressive bom-ref, e.g. the purl (which is also recommended to be used).
This might be very subjective, but at least it would help when manually inspecting the dependencies section.

Reply:
Automatically fixed as I switched to the cyclonedx python package to create the BOM